src/ArmTemplates/Extractor/Models/ExtractorParameters.cs

```diff
@@ -115,6 +116,9 @@ public ExtractorParameters(ExtractorConsoleAppConfiguration extractorConfig)
             this.ExtractGateways = extractorConfig.ExtractGateways != null && extractorConfig.ExtractGateways.Equals("true", StringComparison.OrdinalIgnoreCase);
             this.OverrideGroupGuids = extractorConfig.OverrideGroupGuids != null && extractorConfig.OverrideGroupGuids.Equals("true", StringComparison.OrdinalIgnoreCase);
             this.OverrideProductGuids = extractorConfig.OverrideProductGuids != null && extractorConfig.OverrideProductGuids.Equals("true", StringComparison.OrdinalIgnoreCase);
+            this.ApiParameters = extractorConfig.ApiParameters;
+            this.ParameterizeServiceUrl = extractorConfig.ParamServiceUrl != null && extractorConfig.ParamServiceUrl.Equals("true", StringComparison.OrdinalIgnoreCase) || (extractorConfig.ApiParameters != null && this.ApiParameters.Any(x => x.Value.ServiceUrl is not null));
```

Comment:
It could be treated as valid string even it is empty or have a whitespace. This will allow more flexibility in the configuration file, and not forcing placing exact values there. Also, this approach is used in other parameters. 
I prefer to not change behaviour of parameters for different configurations :) 
Open to discussion :) what do you think?

Reply:
LGTM